lang/stack/Portfile

```diff
@@ -0,0 +1,116 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem          1.0
+PortGroup           github 1.0
+
+github.setup        commercialhaskell stack 2.1.1
+
+name                stack
+categories          lang haskell
+maintainers         {ieee.org:s.t.smith @essandess} openmaintainer
+license             BSD
+platforms           darwin
+
+description The Haskell Tool Stack
+long_description    ${description}. \
+        Stack is a cross-platform command line interface for developing Haskell projects. \
+        It features: installing GHC automatically, installing packages needed for your project, \
+        building your project, testing your project, benchmarking your project. \
+        It is intended for Haskellers both new and experienced.
+
+homepage            https://haskellstack.org
+
+variant bootstrap \
+	    description {Bootstrap a build of stack.} {}
+
+# make bootstrapping the default
+default_variants    +bootstrap
+
+master_sites        ${github.homepage}/releases/download/v${github.version}
+distfiles           ${name}-${github.version}-osx-x86_64${extract.suffix} \
+                    ${name}-${github.version}-osx-x86_64${extract.suffix}.asc
+distname            ${name}-${github.version}-osx-x86_64
+
+checksums           ${name}-${github.version}-osx-x86_64${extract.suffix} \
+                    rmd160  45c8fc4537bf3fdd46854dcd56d51e4c041ac118 \
+                    sha256  f4af329419fb6ee9655b22db04d72a35a5a225e78bdcc605d78334a72c8c2332 \
+                    size    8316888 \
+                    ${name}-${github.version}-osx-x86_64${extract.suffix}.asc \
+                    rmd160  f5f04cdd005e6059b234fb569b296dbd765f2f82 \
+                    sha256  5550d80d5cd1057e6de6d70a0f907e8155f0aa07e45e3a11f914221502eba9f1 \
+                    size    488
+
+extract.only        ${name}-${github.version}-osx-x86_64${extract.suffix}
+
+if {[variant_isset "bootstrap"]} {
+    master_sites-append \
+                    ${github.homepage}/archive
+    distfiles-append \
+                    v${github.version}${extract.suffix}
+    checksums-append \
+                    v${github.version}${extract.suffix} \
+                    rmd160  60f7fcc5548aabec357be217d11d0d1d616c1806 \
+                    sha256  10c0119456748b189724ee42dea093c324d101487c3d75d57eb625675bb57424 \
+                    size    816217
+    extract.only-append \
+                    v${github.version}${extract.suffix}
+}
+
+supported_archs     x86_64
+
+post-checksum {
```

Comment:
Same comment as for the GHC Portfile: I really think the maintainer should do this and the users should rely on the checksums in the Portfile that already provide a chain of trust, rather than starting multiple complex shell commands here.

Reply:
Thanks. If there's consensus not to check author PGP signatures within MacPorts, I'll remove these blocks.

Any other downvotes for signature checking? I'll count @neverpanic and @pmetzger as recommending removal.

Reply:
It's gone. Also see these comments: https://github.com/macports/macports-ports/pull/4825#discussion_r304193581